great_expectations_cloud/agent/agent.py

```diff
@@ -118,17 +118,17 @@ def __init__(self: Self):
         print(f"GX Agent version: {agent_version}")
         print(f"Great Expectations version: {great_expectations_version}")
         print("Initializing the GX Agent.")
-        self._set_http_session_headers()
```

Comment:
refactored to take in `data_context` as parameter and moved it past the `self._context` initialization. behavior should be the same, but allows GX-runner to pass in a DataContext-per job. 

https://github.com/greatexpectationslabs/gx-runner/pull/74